winter_messaging_transactional/tests/tests/test_consumer.py

```diff
@@ -0,0 +1,261 @@
+import json
+from time import sleep
+from uuid import uuid4
+
+import pytest
+from testcontainers.rabbitmq import RabbitMqContainer
+
+from winter.core.json import JSONEncoder
+
+from winter_messaging_transactional.tests.conftest import event_consumer
+from winter_messaging_transactional.tests.helpers import EVENT_HANDLING_TIMEOUT
+from winter_messaging_transactional.tests.helpers import create_rabbitmq_connection
+from winter_messaging_transactional.tests.helpers import get_rabbitmq_url
+from winter_messaging_transactional.tests.helpers import read_all_inbox_messages
+from winter_messaging_transactional.tests.app_sample.dao import ConsumerDAO
+from winter_messaging_transactional.tests.app_sample.events import SampleEvent
+from winter_messaging_transactional.tests.app_sample.events import RetryableEvent
+from winter_messaging_transactional.consumer.inbox.inbox_message import InboxMessage
+from winter_messaging_transactional.consumer.inbox.inbox_message_dao import InboxMessageDAO
+from winter_messaging_transactional.producer.outbox import OutboxEventPublisher
+from winter_messaging_transactional.producer.outbox import OutboxMessage
+from winter_messaging_transactional.producer.outbox import OutboxMessageDAO
+from winter_messaging_transactional.tests.helpers import run_consumer
+from winter_messaging_transactional.tests.helpers import run_processor
+from winter_messaging_transactional.tests.helpers import wait_for_result
+
+
+def test_consume_without_error(database_url, rabbit_url, event_processor, injector, session):
+    event_publisher = injector.get(OutboxEventPublisher)
```

Comment:
nit: why we directly inject injector instead of creating fixture for it and get resolved instance of OutboxEventPublisher in test?